### PR TITLE
Short-circuit the evaluation of `cpuFraction` and `memoryFraction`

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
refactor the code to make it more efficient.

```release-note
NONE
```

### CODE REVIEW

**Comment by chendave** [comment]
pls help to trigger the test, thanks!

File: pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go [hunk 1]
```diff
@@ -42,9 +42,14 @@ func balancedResourceScorer(requested, allocable *schedulernodeinfo.Resource, in
...
+	if cpuFraction >= 1 || memoryFraction >= 1 {
+		// if requested >= capacity, the corresponding host should never be preferred.
+		return 0
+	}
+
 	if includeVolumes && utilfeature.DefaultFeatureGate.Enabled(features.BalanceAttachedNodeVolumes) && allocatableVolumes > 0 {
 		volumeFraction := float64(requestedVolumes) / float64(allocatableVolumes)
-		if cpuFraction >= 1 || memoryFraction >= 1 || volumeFraction >= 1 {
+		if volumeFraction >= 1 {
 			// if requested >= capacity, the corresponding host should never be preferred.
```
Comment (draveness) [inline pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go]: You could remove the comment here

**Review by draveness — CHANGES_REQUESTED** [review]
/ok-to-test
/priority backlog
/sig scheduling

please take a look at the comments, others LGTM

File: pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go (hunk 1 above)
Comment (chendave) [reply pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go]: sure, will update.

**Comment by chendave** [comment]
@draveness thanks for the review, pls take another look, thanks!

**Review by draveness — APPROVED** [review]
/lgtm

Thanks!

**Review by draveness — COMMENTED** [review]
Kindly ping @bsalamat for approval, this is an easy one.

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

Thanks, @chendave!

**Pull request merged**